src/operations/search_indexes/create.ts

```diff
@@ -20,37 +22,36 @@ export interface SearchIndexDescription extends Document {
 }
 
 /** @internal */
-export class CreateSearchIndexesOperation extends AbstractOperation<string[]> {
+export class CreateSearchIndexesOperation extends ModernizedOperation<string[]> {
+  override SERVER_COMMAND_RESPONSE_TYPE = CreateSearchIndexesResponse;
   private readonly collection: Collection;
   private readonly descriptions: ReadonlyArray<SearchIndexDescription>;
 
   constructor(collection: Collection, descriptions: ReadonlyArray<SearchIndexDescription>) {
     super();
     this.collection = collection;
     this.descriptions = descriptions;
+    this.ns = collection.fullNamespace;
   }
 
   override get commandName() {
     return 'createSearchIndexes' as const;
   }
 
-  override async execute(
-    server: Server,
-    session: ClientSession | undefined,
-    timeoutContext: TimeoutContext
-  ): Promise<string[]> {
+  override buildCommand(_connection: Connection, _session?: ClientSession): Document {
     const namespace = this.collection.fullNamespace;
-    const command = {
+    return {
       createSearchIndexes: namespace.collection,
       indexes: this.descriptions
     };
+  }
 
-    const res = await server.command(namespace, command, {
-      session,
-      timeoutContext
-    });
+  override handleOk(response: InstanceType<typeof this.SERVER_COMMAND_RESPONSE_TYPE>): string[] {
+    const indexesCreated = response.indexesCreated?.toObject();
+    return indexesCreated ? Object.entries(indexesCreated).map(([_key, val]) => val.name) : [];
```

Comment:
We have [NODE-6497](https://jira.mongodb.org/browse/NODE-6497) planned for this quarter, which among other things, I expect to figure out how to properly handle deserializing Arrays.  

I think it might be better to hold off on deserializing responses with arrays until then.  what do you think? 

(if we choose to leave this implementation as-is, I think we need to pass `this.bsonOptions` into toObject()).

Reply:
Updated to pass in the bson options.

Reply:
Unless you feel strongly about typing this response, I'd rather wait until we address Type Safe Command Responses.  For now, we can just do:

```
return super.handleOk(response).indexesCreated.map(val => val.name);
```

Nested array fields do not seem trivial to implement, and I'd rather take the time to actually think about a proper solution than just make something work in this PR (i.e., maybe we need an `OnDemandArray` class instead.  and we need a plan for handling BSON deserialization options between parent documents -> subdocuments. etc).